### PR TITLE
 🐛 Fixes multiple clients fixture

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- FIXES: ``aiohttp_client`` creates a server and paired client. The fixtures in test_project02 was generating multiple clients paired with *different* servers. Created a new fixture that creates multiple clients paired with the same server.
- FIXES some type annotations

These changes were performed during upgrade of py3.8 (#2079) but moved to a separate PR for clarity
 
## Related issue/s

This was part of the py3.8 upgrade (PR #2079)

## How to test

Webserver tests cover these changes

```command
make devenv
source .venv/bin/activate
cd services/web/server
make install-dev
make tests-dev

cd ...
cd tests/environment-setup
make install-dev
make tests
```

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [ ] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/web/server/tests/unit/with_dbs/06/test_projects_02.py [hunk 1]
```diff
@@ -908,16 +909,53 @@ async def test_tags_to_studies(
...
+    # Creates clients connected to the same server as the reference client
+    #
+    # Implemented as aihttp_client but creates a client using a running server,
+    #  i.e. avoid client.start_server
+
+    assert isinstance(client.server, TestServer)
+
+    clients = []
+
+    def go():
```
Comment (GitHK) [inline services/web/server/tests/unit/with_dbs/06/test_projects_02.py:923]: maybe a more meaningful name would help here

File: tests/environment-setup/test_used_python.py [hunk 2]
```diff
@@ -39,8 +37,20 @@ def make_versions_comparable(*versions) -> List[Tuple[int]]:
     return [v[:n] for v in vers]
 
 
+@pytest.fixture(scope="session")
+def expected_python_version(osparc_simcore_root_dir: Path) -> Tuple[int, ...]:
```
Comment (GitHK) [inline tests/environment-setup/test_used_python.py:41]: I do not understand the idea behind this test

Comment (pcrespov) [reply tests/environment-setup/test_used_python.py:41]: this is a fixture (not a test)
it is responsible to identify the main python version of the repo
it is very convenient to do python upgrades
remember that this is a part of the py3.8 PR

File: services/web/server/tests/unit/with_dbs/06/test_projects_02.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/tests/unit/with_dbs/06/test_projects_02.py:923]: haha ... this is a modification of ``aiohttp.pytest_plugin`` 
It is a standard used in all libraries under https://github.com/aio-libs. You find it everywhere.
``go`` to initialize something and ``finalize`` to tear-down

Comment (GitHK) [reply services/web/server/tests/unit/with_dbs/06/test_projects_02.py:923]: I need context to comprehend the context. I see! 🤣

**Review by GitHK — APPROVED** [review]

**Review by mguidon — APPROVED** [review]

**Pull request merged**